src/app/libraries/http_lib.php

```diff
@@ -1,14 +1,19 @@
 <?php
 
 class http_lib extends Library {
-
     function response_code($code, $view_and_exit = true, $view_name = false) {
         http_response_code($code);
         if ($view_and_exit) {
             if (empty($view_name)) {
                 $view_name = strval($code);
             }
-            $this->load_view($view_name);
+            // Check if a dedicated view exists; if not, load common error view
+            $dedicatedView = $this->load_view($code);
+            if (! $dedicatedView) {
+                $this->load_view('http_error', [
+                    'errorCode' => $code,
```

Comment:
Like I reasoned above, this will only happen in the case of errors. I don't see the need to make it status code.